File: valid_braces/valid_braces.java
import java.util.Map;
import java.util.Stack;

public class BraceChecker {
    public boolean isValid(String braces) {
        final Map r_bracket_pairs = Map.of(')', '(', ']', '[', '}', '{');
        Stack<Character> braces_stack = new Stack<>();
        for (char brace : braces.toCharArray()) {
            switch (brace) {
            case '(', '[', '{' -> braces_stack.push(brace);
            case ')', ']', '}' -> {
                if (braces_stack.empty() || braces_stack.pop() != r_bracket_pairs[braces]) {
                    return false;
                }
            }
            }
        }
        return braces_stack.empty();
    }
}

File: valid_braces/valid_braces.go
package kata

type stack []rune

func (s stack) Push(v rune) stack {
	return append(s, v)
}

func (s stack) Top() rune {
	length := len(s)
	return s[length-1]
}

func (s stack) Pop() stack {
	length := len(s)
	return s[:length-1]
}

func (s stack) Empty() bool {
	return len(s) == 0
}

func ValidBraces(str string) bool {
	r_bracket_pairs := map[rune]rune{
		')': '(',
		']': '[',
		'}': '{',
	}
	stack := make(stack, 0)
	for _, s := range str {
		switch s {
		case '(', '[', '{':
			stack = stack.Push(s)
		case ')', ']', '}':
			if stack.Empty() || stack.Top() != r_bracket_pairs[s] {
				return false
			}
			stack = stack.Pop()
		}
	}
	return stack.Empty()
}
